### ISSUE TITLE
False-positive "used-before-assignment" on ternary variable assignment (python3.8)

### ISSUE THREAD

**Issue by Mastermind-U** [PyCQA/pylint#3763]
### Steps to reproduce
1. Python3.8 works fine:
```python3
>>> foo if (foo := 3 - 2) > 0 else 0
1
```
2. Run pylint:
```bash 
pylint somefile.py
```

### Current behavior
Got output `E0601: Using variable 'foo' before assignment (used-before-assignment)`

### pylint --version output
```bash
pylint 2.5.3
astroid 2.4.2
Python 3.8.3 (default, May 17 2020, 18:15:42) 
[GCC 10.1.0]
```

**Comment by AWhetter** [comment]
I can reproduce this on master. As far as I'm aware, we don't have very good support for walrus operators yet.
